torch/csrc/jit/codegen/cuda/lower_predicate_elimination.cpp

```diff
@@ -450,8 +538,24 @@ class PredicateChcker : public IterVisitor {
 
       if (non_predicated_exprs_.find(input_def) !=
           non_predicated_exprs_.end()) {
-        needs_predicate_ = true;
-        return;
+        // If producer of mma is non_predicated and initialized
+        //  with the same value. The mma should not need a
+        //  predicate. In fact this is the only way we can
+        //  use mma at the moment since we could not predicate
+        //  mma ops without guaranteeing warp uniform results.
+        if (mma->fusion()->unordered_uses(input).size() > 1) {
```

Comment:
Why `> 1`?

Reply:
Changed to checking initial values and could merge this thread with the one above.